src/net.h

```diff
@@ -344,6 +344,40 @@ class V1TransportSerializer : public TransportSerializer {
     void prepareForTransport(CSerializedNetMsg& msg, std::vector<unsigned char>& header) const override;
 };
 
+/**
+ * Placeholder for total network traffic. Split by direction, network, connection
+ * type and message type (byte and message counts).
+ */
+class NetStats
+{
+public:
+    struct MsgStat {
+        std::atomic_uint64_t byte_count; //!< Number of bytes transferred.
+        std::atomic_uint64_t msg_count;  //!< Number of messages transferred.
+
+        MsgStat() = default;
+
+        MsgStat(const MsgStat& x) : byte_count{x.byte_count.load()}, msg_count{x.msg_count.load()} {}
+
+        MsgStat(std::atomic_uint64_t x, std::atomic_uint64_t y) : byte_count{x.load()}, msg_count{y.load()} {}
+    };
+
+    enum class Direction { SENT,
+                            RECV };
+
+    /// Number of elements in `Direction`.
+    static constexpr size_t NUM_DIRECTIONS{2};
+
+    using MultiDimensionalStats = std::array<std::array<std::array<std::array<MsgStat,
+                                                                                // add 1 for the other message type
+                                                                                NUM_NET_MESSAGE_TYPES + 1>,
+                                                                    NUM_CONNECTION_TYPES>,
+                                                        NET_MAX>,
+                                                NUM_DIRECTIONS>;
```

Comment:
oof. some of the spacing is not quite right here even for cascading. I'm going to fix it in the new compilation unit/file, maybe then it will make a little more sense? I'm worried lining them up isn't going to be much help. It's hard to read too because the declaration for this 4D array happens by nesting arrays in arrays :sob: